### PR TITLE
Add flag for subscription based notification

### PR DESCRIPTION
**Pull request by rish9101**

### CODE REVIEW

File: actions.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import logging
 
+from categories.models import UserSubscription
+
```
Comment (pradumangoyal) [inline actions.py]: This line break is not required

File: actions.py [hunk 2]
```diff
@@ -100,10 +104,19 @@ def push_notification(
             )
         else:
             notification.save()
+
+            if send_only_to_subscribed_targets:
+
```
Comment (pradumangoyal) [inline actions.py]: Remove this break too

**Review by pradumangoyal — COMMENTED** [review]
There is an open PR #6 which supports Person instance as well as ID. So, we may first merge them and have the required changes here.

File: actions.py [hunk 3]
```diff
@@ -100,10 +103,18 @@ def push_notification(
             )
         else:
             notification.save()
+
+            if send_only_to_subscribed_targets:
+                persons = UserSubscription.objects.filter(
+                    category=category
+                ).filter(person__id__in=persons)\
```
Comment (Ajayneethikannan) [inline actions.py]: Not sure of this, but would default persons=None throw an error here? Might be better to use an empty array

Comment (pradumangoyal) [reply actions.py]: Actually, we should fetch it from redisdb like this: https://github.com/IMGIITRoorkee/omniport-service-emails/blob/3a98f640d4578372519cc4f34ea3f1fd2ab112cd/actions.py#L74

Comment (pradumangoyal) [reply actions.py]: And then we can take an intersection of both.

**Pull request merged**